src/transactions/TransactionFrame.cpp

```diff
@@ -1626,15 +1625,15 @@ TransactionFrame::checkValidWithOptionallyChargedFee(
         getSignatures(mEnvelope)};
 
     std::optional<FeePair> sorobanResourceFee;
-    std::optional<SorobanNetworkConfig> sorobanConfig;
+    SorobanNetworkConfig const* sorobanConfig = nullptr;
```

Comment:
Using a raw pointer feels like a footgun. Would it make sense to update the ledger state snapshot to store the config in a shared_ptr instead of an optional?

Reply:
Pointer is just a nullable reference (or reference is just a non-nullable pointer) and it's not inherently more or less footgun-y than const references that we're using everywhere without much issue. There is no ownership sharing intended here, so I don't think shared ptr is appropriate. 
I've tried to use optional where it was feasible, but for this case I don't want to make getLastClosedSorobanNetworkConfig optional (or, generally, any nullable object, as it's not nullable), and I also don't want to create a new object every time we perform validation. So this seems like the most reasonable tradeoff.